src/libultra/os/writebackdcache.s

```diff
@@ -16,45 +12,43 @@
  *  written back.
  */
 LEAF(osWritebackDCache)
-    // If the amount to write back is less than or equal to 0, return immediately
-    blez    $a1, .ret
-     nop
-    // If the amount to write back is as large as or larger than
-    // the data cache size, write back all
-    li      $t3, DCACHE_SIZE
-    sltu    $at, $a1, $t3
-    beqz    $at, .all
-     nop
-    // ensure end address doesn't wrap around and end up smaller
-    // than the start address
-    move    $t0, $a0
-    addu    $t1, $a0, $a1
-    sltu    $at, $t0, $t1
-    beqz    $at, .ret
-     nop
-    // Mask and subtract to align to cache line
-    andi    $t2, $t0, DCACHE_LINEMASK
-    addiu   $t1, $t1, -DCACHE_LINESIZE
-    subu    $t0, $t0, $t2
+    /* If the amount to write back is less than or equal to 0, return immediately */
+    blez    a1, .ret
+
+    /* If the amount to write back is as large as or larger than */
+    /* the data cache size, write back all */
+    li      t3, DCACHE_SIZE
+    bgeu    a1, t3, .all
+
+    /* ensure end address doesn't wrap around and end up smaller */
+    /* than the start address */
+    move    t0, a0
+    addu    t1, a0, a1
+    bgeu    t0, t1, .ret
+
+    /* Mask and subtract to align to cache line */
+    andi    t2, t0, DCACHE_LINEMASK
+    addiu   t1, t1, -DCACHE_LINESIZE
+    subu    t0, t0, t2
 1:
-    cache   (CACH_PD | C_HWB), ($t0)
-    sltu    $at, $t0, $t1
-    bnez    $at, 1b
-     addiu  $t0, $t0, DCACHE_LINESIZE
+        .set noreorder
```

Comment:
dedent `.set noreorder` and `.set reorder`? ditto below and in writebackdcacheall